camel/typing.py

```diff
@@ -77,8 +80,10 @@ def token_limit(self) -> int:
             return 4096
         elif self is ModelType.GPT_3_5_TURBO_16K:
             return 16384
+        elif self is ModelType.GPT_3_5_TURBO_1106:
+            return 4096 
         elif self is ModelType.GPT_4:
-            return 8192
+            return 128000
```

Comment:
`"gpt-4"` 's context window is still 8,192 tokens. GPT-4 Turbo's and GPT-4 Turbo with vision's context window are 128,000 tokens.